plugins/outputs/sumologic/sumologic.go

```diff
@@ -0,0 +1,307 @@
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/serializers"
+	"github.com/influxdata/telegraf/plugins/serializers/carbon2"
+	"github.com/influxdata/telegraf/plugins/serializers/graphite"
+	"github.com/influxdata/telegraf/plugins/serializers/prometheus"
+)
+
+const (
+	sampleConfig = `
+# A plugin that can send metrics to Sumo Logic HTTP metric collector.
+[[outputs.sumologic]]
+  ## Unique URL generated for your HTTP Metrics Source.
+  ## This is the address to send metrics to.
+  # url = "https://events.sumologic.net/receiver/v1/http/<UniqueHTTPCollectorCode>"
+
+  ## Data format to be used for sending metrics.
+  ## This will set the "Content-Type" header accordingly.
+  ## Currently supported formats: 
+  ## * graphite - for Content-Type of application/vnd.sumologic.graphite
+  ## * carbon2 - for Content-Type of application/vnd.sumologic.carbon2
+  ## * prometheus - for Content-Type of application/vnd.sumologic.prometheus
+  ##
+  ## More information can be found at:
+  ## https://help.sumologic.com/03Send-Data/Sources/02Sources-for-Hosted-Collectors/HTTP-Source/Upload-Metrics-to-an-HTTP-Source#content-type-headers-for-metrics
+  ##
+  ## NOTE:
+  ## When unset, telegraf will by default use the influx serializer which is currently unsupported
+  ## in HTTP Source.
+  data_format = "carbon2"
+
+  ## Timeout used for HTTP request
+  # timeout = "5s"
+  
+  ## HTTP method, one of: "POST" or "PUT". "POST" is used by default if unset.
+  # method = "POST"
+
+  ## HTTP Content-Encoding for writing request body. Can be set to "gzip" to compress the body.
+  ## This will set the "Content-Type" header accordingly.
+  # content_encoding = "gzip"
+
+  ## Max HTTP request body size in bytes before compression (if applied).
+  ## By default 1MB is recommended.
+  ## NOTE:
+  ## Bear in mind that in some serializer a metric even though serialized to multiple
+  ## lines cannot be split any further so setting this very low might not work
+  ## as expected.
+  # max_request_body_size = 1_000_000
+
+  ## Additional, sumologic specific HTTP header that can be set.
+  ## Full list can be found here:
+  ## https://help.sumologic.com/03Send-Data/Sources/02Sources-for-Hosted-Collectors/HTTP-Source/Upload-Metrics-to-an-HTTP-Source#supported-http-headers
+  [outputs.sumologic.headers]
+    ## Desired source name
+    # X-Sumo-Name = ""       
+    ## Desired host name
+    # X-Sumo-Host = ""
+    ## Desired source category
+    # X-Sumo-Category  = ""
+    ## Comma-separated key=value list of dimensions to apply to every metric.
+    # X-Sumo-Dimensions = ""
+    ## Comma-separated, key=value list of metadata to apply to every metric.
+    # X-Sumo-Dimensions = ""
+`
+
+	defaultClientTimeout      = 5 * time.Second
+	defaultContentType        = ""
+	defaultMethod             = http.MethodPost
+	defaultMaxRequestBodySize = 1_000_000
```

Comment:
Might be good to check with metrics and collecton teams on optimzal batch sizes.

Reply:
I followed the hint from (here](https://help.sumologic.com/03Send-Data/Sources/02Sources-for-Hosted-Collectors/HTTP-Source#data-payload-considerations) where it says that it's best the batches are between 100KB and 1MB so I just wanted to make it smaller than the max.

But sure, I'll talk to them.

Reply:
ahh perfect, if you did that i think we are good and assuming we will compress the transmission with GZIP as well?

Reply:
I can't think of a reason we would want to make that configurable. even our installed collector always compresses.

Reply:
What exactly do you refer here @frankreno ? Configuration whether to compress it via gzip (should we make that enabled  by default?) or the max request body size?

Reply:
Sorry was not very clear.  So for the request size, I am fine with exposing that for fine-tuning in cases where it may make sense. However, GZIP compression should not be configurable as I can think of a good reason we would not want to compress the data.

Reply:
Done. I've made `gzip` default and I've hidden to option to change it.